<?xml version="1.0"?>
<launch>
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  
  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg robot_description)_semantic" textfile="$(find moveo_moveit_config)/config/moveo_urdf.srdf" />

  <!-- Load default settings for kinematics; these settings are overridden by settings in a node's namespace -->
  <group ns="$(arg robot_description)_kinematics">
    <rosparam command="load" file="$(find moveo_moveit_config)/config/kinematics.yaml"/>
  </group>
  <arg name="chain_start" default="base_link" />
  <arg name="chain_end" default="moveo_nozzle_link" />
  <arg name="timeout" default="0.15" />
  <arg name="timeout_second" default="1" />
  <arg name="eps" default="1e-5"/>
  <arg name="X_offset" default="-0.027"/>
  <arg name="Y_offset" default="-0.007"/>
  <arg name="Z_offset" default="0.10"/>
  <!--
    arg name="num_threads" default="16"/
    -->
  <arg name="model" default="$(find moveo_urdf)/urdf/moveo_urdf1.urdf"/>
  <arg name="model2" default="$(find moveo_urdf)/urdf/moveo_urdf2.urdf"/>
  <param name="robot_description" textfile="$(arg model)" />
  <param name="robot_description2" textfile="$(arg model2)" />
  
  <arg name="g_n_judge_Y" default="$(find gcode_translation)/p_n_judge_Y/p_n_judge_Y.txt"/>
  <arg name="gcode_in" default="$(find gcode_translation)/gcode/split_register.gcode"/>
  <arg name="gcode_out" default="$(find gcode_translation)/gcode/new_Joint_XYZ.gcode"/>
  <arg name="check_success" default="$(find gcode_translation)/check_success/check_success.txt"/>
  <arg name="node_start_delay" default="1.0" />
  <node name="inverse_kinematics" pkg="gcode_translation" type="inverse_kinematics" output="screen" required="true" >
    <param name="chain_start" value="$(arg chain_start)"/>
    <param name="chain_end" value="$(arg chain_end)"/>
    <param name="timeout" value="$(arg timeout)"/>
    <param name="timeout_second" value="$(arg timeout_second)"/>
    <param name="eps" value="$(arg eps)" type = "double"/>
    <param name="X_offset" value="$(arg X_offset)" type = "double"/>
    <param name="Y_offset" value="$(arg Y_offset)" type = "double"/>
    <param name="Z_offset" value="$(arg Z_offset)" type = "double"/>
    <!--
      param name="num_threads" value="$(arg num_threads)" type = "int"/
      -->
    <param name="urdf_param" value="/robot_description"/>
    <param name="urdf_param2" value="/robot_description2"/>
    <param name="g_n_judge_Y" value="$(arg g_n_judge_Y)"/>
    <param name="gcode_in" value="$(arg gcode_in)"/>
    <param name="gcode_out" value="$(arg gcode_out)"/>
    <param name="check_success" value="$(arg check_success)"/>
  </node>
  <include file="$(find progressbar)/launch/progressbar.launch" />
</launch>
